Remotion/Web/Development.WebTesting/Utilities/AssertionExceptionUtility.cs

```diff
@@ -28,62 +29,69 @@ public static class AssertionExceptionUtility
+    public static WebTestException CreateCommandDisabledException ([CallerMemberName] string operationName = "", IDriver driver = null)
     {
       ArgumentUtility.CheckNotNullOrEmpty ("operationName", operationName);
 
-      return new WebTestException (string.Format ("The command is currently in a disabled state. Therefore, the '{0}' operation is not possible.", operationName));
+      return CreateException (string.Format ("The command is currently in a disabled state. Therefore, the '{0}' operation is not possible.", operationName), driver);
     }
 
     [NotNull]
     [MustUseReturnValue]
-    public static WebTestException CreateControlReadOnlyException ()
+    public static WebTestException CreateControlReadOnlyException (IDriver driver = null)
     {
-      return new WebTestException ("The control is currently in a read-only state. Therefore, the operation is not possible.");
+      return CreateException ("The control is currently in a read-only state. Therefore, the operation is not possible.", driver);
     }
 
     [NotNull]
     [MustUseReturnValue]
-    public static WebTestException CreateControlNotReadOnlyException ()
+    public static WebTestException CreateControlNotReadOnlyException (IDriver driver = null)
     {
-      return new WebTestException ("The control is currently not in a read-only state. Therefore, the operation is not possible.");
+      return CreateException ("The control is currently not in a read-only state. Therefore, the operation is not possible.", driver);
     }
 
     [NotNull]
     [MustUseReturnValue]
-    public static WebTestException CreateControlMissingException ([NotNull] string exceptionDetails)
+    public static WebTestException CreateControlMissingException ([NotNull] string exceptionDetails, IDriver driver = null)
     {
       ArgumentUtility.CheckNotNullOrEmpty ("exceptionDetails", exceptionDetails);
 
-      return new WebTestException ($"The element cannot be found: {exceptionDetails}");
+      return CreateException ($"The element cannot be found: {exceptionDetails}", driver);
     }
 
     [NotNull]
     [MustUseReturnValue]
-    public static WebTestException CreateControlAmbiguousException ([NotNull] string exceptionDetails)
+    public static WebTestException CreateControlAmbiguousException ([NotNull] string exceptionDetails, IDriver driver = null)
     {
       ArgumentUtility.CheckNotNullOrEmpty ("exceptionDetails", exceptionDetails);
 
-      return new WebTestException ($"Multiple elements were found: {exceptionDetails}");
+      return CreateException ($"Multiple elements were found: {exceptionDetails}", driver);
     }
 
     [NotNull]
     [MustUseReturnValue]
     [StringFormatMethod ("message")]
-    public static WebTestException CreateExpectationException ([NotNull] string message, params object[] args)
+    public static WebTestException CreateExpectationException ([NotNull] string message, IDriver driver = null, params object[] args)

```

Comment:
format and args shouldn't be split. we need the driver to be the first, and then it can't be optional. since we always want a driver, that shouldn't be an issue, correct? this applies to all factory methods for consistency reasons. also are there downstream-dependencies?